src/admin/admin.service.ts

```diff
@@ -34,8 +35,19 @@ export class AdminService {
                     decryptionCode: Math.random().toString(36).substring(2, 15) + Math.random().toString(36).substring(2, 15),
                     ...userInput
                 });
+                if (userInput.id_code === "")
+                    continue;
+                await fetch("http://infra-cdocweb-1/cdoc", {
+                    method: 'POST',
+                    body: generatedPath[i] + "&" + generatedPath[nextIndex] + "&" + userInput.id_code,
+                    headers: {'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8'} }
+                ).then((result) => {
+                    return result.text();
+                }).then((data) => {
+                    fs.writeFileSync("public/cdoc_files/"+generatedPath[i]+".cdoc", data);
```

Comment:
The files should not be public, but should be served based on user authentication through a special endpoint.